src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+import core.basesyntax.exception.UserLoginExistsException;
+import core.basesyntax.exception.UserLoginLengthException;
+import core.basesyntax.exception.UserPasswordLengthException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    @DisplayName("User register with null login")
+    void register_nullLogin_throwNullPointerException() {
+        User userWithNullLogin = new User(null, "1234567", 45);
+        assertThrows(NullPointerException.class,
+                     () -> registrationService.register(userWithNullLogin));
+    }
+
+    @Test
+    @DisplayName("User register with null password")
+    void register_nullPassword_throwNullPointerException() {
+        User userWithNullPassword = new User("ivanivanov12@gmail.com", null, 45);
+        assertThrows(NullPointerException.class,
+                     () -> registrationService.register(userWithNullPassword));
+    }
+
+    @Test
+    @DisplayName("User register with null age")
+    void register_nullAge_throwNullPointerException() {
+        User userWithNullAge = new User("ivanivanov12@gmail.com", "1234567", null);
+        assertThrows(NullPointerException.class,
+                     () -> registrationService.register(userWithNullAge));
+    }
+
+    @Test
+    @DisplayName("User register correct user case #1")
+    void register_correctUser_ok() {
+        User user = new User("mykhailo777@gmail.com", "1234567", 23);
+        User actualUser = registrationService.register(user);
+        User expectedUser = new User("mykhailo777@gmail.com", "1234567", 23);
+        assertEquals(actualUser, expectedUser);
+    }
+
+    @Test
+    @DisplayName("User register correct user case #2")
+    void register_correctUserSecondCase_ok() {
+        User user = new User("ivangolybev@gmail.com", "1234567", 28);
+        User actualUser = registrationService.register(user);
+        User expectedUser = new User("ivangolybev@gmail.com", "1234567", 28);
+        assertEquals(actualUser, expectedUser);
+    }
+
+    @Test
+    @DisplayName("User register correct user case #3")
+    void register_correctUserThirdCase_ok() {
+        User user = new User("oleksiiivanov@gmail.com", "1234567", 33);
+        User actualUser = registrationService.register(user);
+        User expectedUser = new User("oleksiiivanov@gmail.com", "1234567", 33);
+        assertEquals(actualUser, expectedUser);
+    }
+
+    @Test
+    @DisplayName("User login already exists register")
+    void register_userLoginExist_throwUserLoginExistException() {
+        User user = new User("mykhailo88525@gmail.com", "1234567", 23);
+        User actualUser = registrationService.register(user);
+        assertThrows(UserLoginExistsException.class,
+                     () -> registrationService.register(actualUser));
```

Comment:
This test case is flawed because it assumes `register` method works correctly by using it to add a user before testing if it throws an exception for a duplicate login. Use a low-level method to add a user to the storage for test setup instead.